src/command/expire/expire.c

```diff
@@ -0,0 +1,680 @@
+
+                    for (unsigned int archiveIdx = 0; archiveIdx < strLstSize(listArchiveDisk); archiveIdx++)
+                    {
+                        // Is there an archive directory for this archvieId?
+                        if (strCmp(archiveId, strLstGet(listArchiveDisk, archiveIdx)) != 0)
+                            continue;
+
+                        StringList *archiveSplit = strLstNewSplitZ(archiveId, "-");
+                        unsigned int archivePgId = cvtZToUInt(strPtr(strLstGet(archiveSplit, 1)));
+
+                        // From the global list of backups to retain, create a list of backups, oldest to newest, associated with
+                        // this archiveId (e.g. 9.4-1), e.g. If globalBackupRetention has 4F, 3F, 2F, 1F then
+                        // localBackupRetentionList will have 1F, 2F, 3F, 4F (assuming they all have same history id)
+                        for (unsigned int retentionIdx = strLstSize(globalBackupRetentionList) - 1;
+                            (int)retentionIdx >=0; retentionIdx--)
+                        {
+                            for (unsigned int backupIdx = 0; backupIdx < infoBackupDataTotal(infoBackup); backupIdx++)
+                            {
+                                InfoBackupData backupData = infoBackupData(infoBackup, backupIdx);
+                                if ((strCmp(backupData.backupLabel, strLstGet(globalBackupRetentionList, retentionIdx)) == 0) &&
+                                    (backupData.backupPgId == archivePgId))
+                                {
+                                    strLstAdd(localBackupRetentionList, backupData.backupLabel);
+                                }
+                            }
+                        }
+
+                        // If no backup to retain was found
+                        if (strLstSize(localBackupRetentionList) == 0)
+                        {
+                            // If this is not the current database, then delete the archive directory else do nothing since the
+                            // current DB archive directory must not be deleted
+                            InfoPgData currentPg = infoPgDataCurrent(infoArchivePgData);
+                            if (currentPg.id != archivePgId)
+                            {
+                                String *fullPath = storagePath(
+                                    storageRepo(), strNewFmt(STORAGE_REPO_ARCHIVE "/%s", strPtr(archiveId)));
+                                storagePathRemoveP(storageRepoWrite(), fullPath, .recurse = true);
+                                LOG_INFO("remove archive path: %s", strPtr(fullPath));
+                            }
+
+                            // Continue to next directory
+                            break;
+                        }
+
+                        // If get here, then a local backup was found for retention
```

Comment:
If we get here?

Reply:
We never use pronouns in the comments which is why it is omitted.